doc/source/contributing.rst

```diff
@@ -4,13 +4,574 @@
+
+Here `origin` is the default name given to your remote repository on GitHub.
+You can see the remote repositories ::
+
+    git remote -v
+
+If you added the upstream repository as described above you will see something
+like ::
+
+    origin  git@github.com:yourname/pandas.git (fetch)
+    origin  git@github.com:yourname/pandas.git (push)
+    upstream        git://github.com/pydata/pandas.git (fetch)
+    upstream        git://github.com/pydata/pandas.git (push)
+
+Now your code is on GitHub, but it is not yet a part of the *pandas* project.  For that to 
+happen, a Pull Request needs to be submitted on GitHub.
+
+Contributing your changes to *pandas*
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Review your code
+----------------
+
+When you're ready to ask for a code review, you will file a Pull Request. Before you do, 
+again make sure you've followed all the guidelines outlined in this document regarding 
+code style, tests, performance tests, and documentation. You should also double check 
+your branch changes against the branch it was based off of:
+
+#. Navigate to your repository on GitHub--https://github.com/your-user-name/pandas.
+#. Click on `Branches`.
+#. Click on the `Compare` button for your feature branch.
+#. Select the `base` and `compare` branches, if necessary. This will be `master` and 
+   `shiny-new-feature`, respectively.
+
+Finally, make the Pull Request
+------------------------------
+
+If everything looks good you are ready to make a Pull Request.  A Pull Request is how 
+code from a local repository becomes available to the GitHub community and can be looked 
+at and eventually merged into the master version.  This Pull Request and its associated 
+changes will eventually be committed to the master branch and available in the next 
+release.  To submit a Pull Request:
+
+#. Navigate to your repository on GitHub.
+#. Click on the `Pull Request` button.
+#. You can then click on `Commits` and `Files Changed` to make sure everything looks okay one last time.
+#. Write a description of your changes in the `Preview Discussion` tab.
+#. Click `Send Pull Request`.
+
+This request then appears to the repository maintainers, and they will review
+the code. If you need to make more changes, you can make them in
+your branch, push them to GitHub, and the pull request will be automatically
+updated.  Pushing them to GitHub again is done by::
+
+    git push -f origin shiny-new-feature
+
+This will automatically update your Pull Request with the latest code and restart the Travis-CI tests.
+
+Delete your merged branch (optional)
+------------------------------------
+
+Once your feature branch is accepted into upstream, you'll probably want to get rid of 
+the branch. First, merge upstream master into your branch so git knows it is safe to delete your branch ::
+
+    git fetch upstream
+    git checkout master
+    git merge upstream/master
+
+Then you can just do::
+
+    git branch -d shiny-new-feature
```

Comment:
this deletes the local branch, but not the branch on github.
There's a button for that on the botton of the PR page , or `push :branch`.